stac_generator/raster_data/generator.py

```diff
@@ -0,0 +1,133 @@
+        
+        try:
+            with rasterio.open(source_cfg.location) as src:
+                bounds = src.bounds
+                bbox = [bounds.left, bounds.bottom, bounds.right, bounds.top]
+                transform = src.transform
+                crs = src.crs
+                
+                geometry = {
+                    "type": "Polygon",
+                    "coordinates": [[
+                        [bounds.left, bounds.bottom],
+                        [bounds.left, bounds.top],
+                        [bounds.right, bounds.top],
+                        [bounds.right, bounds.bottom],
+                        [bounds.left, bounds.bottom]
+                    ]]
+                }
+
+                # Create base item
+                item = pystac.Item(
+                    id=source_cfg.prefix,
+                    geometry=geometry,
+                    bbox=bbox,
+                    datetime=source_cfg.datetime,
+                    properties={}  # properties will be managed by extensions
+                )
+
+                # Initialize EOExtension and set eo:bands as Band objects
+                eo_ext = EOExtension.ext(item, add_if_missing=True)
+                eo_bands = []
+                for idx, band_info in enumerate(source_cfg.bands, start=1):
+                    band = Band.create(
+                        name=f"B{idx}",
+                        common_name=BAND_MAPPING.get(band_info.name.lower(), None),
+                        center_wavelength=float(band_info.wavelength) if band_info.wavelength not in [None, "no band specified"] else None
+                    )
+                    eo_bands.append(band)
+
+                # Ensure eo:bands is managed by EOExtension only
+                eo_ext.bands = eo_bands
+
+                # Update other properties
+                item.properties.update({
+                    "gsd": float(src.res[0]),
+                    "platform": "UAV",
+                    "instruments": ["Multispectral Camera"],
+                })
+
+                # Add projection extension
+                proj_ext = ProjectionExtension.ext(item, add_if_missing=True)
+                if source_cfg.epsg != crs.to_epsg():
+                    raise ValueError(f"EPSG mismatch: {source_cfg.epsg} vs {crs.to_epsg()}")
+                proj_ext.epsg = crs.to_epsg()
+                proj_ext.transform = list(transform)[:6]
+                proj_ext.shape = [src.height, src.width]
+
+                # Add the asset
+                media_type = pystac.MediaType.COG if src.driver.upper() == 'COG' else pystac.MediaType.GEOTIFF
+                asset = pystac.Asset(
+                    href=str(source_cfg.location),
+                    media_type=media_type,
+                    roles=["data"],
+                    title="Raster Data"
+                )
+                item.add_asset("data", asset)
+
+                # Add STAC extensions
+                item.stac_extensions = [
```

Comment:
Its been a while since I wrote the prototype but Im pretty sure this should be populated automatically by pystac. Did you use the existing GeotiffGenerator class as a reference? There are many things that code does, or comments it contains that are not addressed here.

Reply:
correct - removed in latest update